### PR TITLE
Implement WgpuAutoGui mixin to avoid duplicating code

### LINKED ISSUE #230: Fix tri-plication of event handler code in auto module

### PR DESCRIPTION
**Pull request by almarklein**
Closes #230

### CODE REVIEW

File: wgpu/gui/base.py [hunk 1]
```diff
@@ -175,3 +175,72 @@ def _request_draw(self):
...
+    def handle_event(self, event):
+        """Handle an incoming event.
+
+        Subclasses can overload this method. Events include widget
+        resize, mouse/touch interaction, key events, and more. An event
+        is a dict with at least the key event_type. For details, see
+        https://jupyter-rfb.readthedocs.io/en/latest/events.html
+        """
+        event_type = event.get("event_type")
+        for callback in self._event_handlers[event_type]:
```
Comment (Korijn) [inline wgpu/gui/base.py:192]: Are you ok with assuming `self._event_handlers` exists?

File: wgpu/gui/offscreen.py [hunk 2]
```diff
@@ -3,9 +3,10 @@
 import numpy as np
 
 from ._offscreen import WgpuOffscreenCanvas
+from .base import WgpuAutoGui
 
 
-class WgpuManualOffscreenCanvas(WgpuOffscreenCanvas):
+class WgpuManualOffscreenCanvas(WgpuAutoGui, WgpuOffscreenCanvas):
```
Comment (Korijn) [inline wgpu/gui/offscreen.py:9]: I don't think this one has `self._event_handlers` now

File: wgpu/gui/base.py (hunk 1 above)
Comment (almarklein) [reply wgpu/gui/base.py:198]: No, I'm not ;)

File: wgpu/gui/offscreen.py (hunk 2 above)
Comment (almarklein) [reply wgpu/gui/offscreen.py:9]: Good catch.

**Pull request merged**